### PR TITLE
RFP:ars-go-filecoin-storage-helper by ARS team

### PR DESCRIPTION
**Pull request by waynewyang**
Name of Project: ars-go-filecoin-storage-helper

### CODE REVIEW

File: rfp-proposals/ars-go-filecoin-storage-helper.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+| 1 | Storage helper library | $3500 | 2 weeks |
+| 2 | All deliverables (codebase, sample application, documentation, testing) | $1500 | 1 week |
+
+
+## Total Budget Requested
+
+**Total Funding Amount:** $5000
+
+
+## Maintenance and Upgrade Plans
```
Comment (pooja) [inline rfp-proposals/ars-go-filecoin-storage-helper.md]: Can you clarify whether you are planning to maintain this project on your own (i.e. without additional funding)?

File: rfp-proposals/ars-go-filecoin-storage-helper.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**  "Yes"
+
+# Project Description
+
+The project is based on [rfp-storage-helper-libraries](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md), We think we're  very clear about the project want to do, our implementation will meet the need of [RFP](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md),
+
+We think the keypoint is that client must pay for the metadata, the metadata is used for data retrieval,so the design of the metadata is very important,now we already have the idea.
+
+Besides,we have another idea we'll provide the function in storage helper that metadata encryption.
+
```
Comment (pooja) [inline rfp-proposals/ars-go-filecoin-storage-helper.md]: Can you describe how you expect this library to be used? Does it have a command-line interface? If so, what are some of the commands? Can we make it work with lotus?

File: rfp-proposals/ars-go-filecoin-storage-helper.md (hunk 1 above)
Comment (waynewyang) [reply rfp-proposals/ars-go-filecoin-storage-helper.md]: @pooja  yes,we'll maintenance work without additional funding.

File: rfp-proposals/ars-go-filecoin-storage-helper.md (hunk 2 above)
Comment (waynewyang) [reply rfp-proposals/ars-go-filecoin-storage-helper.md]: - You can use by import the package or use command, you can view the workflow [here](https://github.com/arsyun/go-filecoin-storage-helper)
- Yes, such as 
    - storage-helper store [path]
        - --status [cid] // check the storage flowwork status
        - --expire [time] // set expire time
        - --price [FIL] // set price
        - ......
    - storage-helper retrieval
        - ......
- We'll support lotus although lotus was not open source when the proposal was submitted

Comment (waynewyang) [reply rfp-proposals/ars-go-filecoin-storage-helper.md]: - You can use by import the package or use command, you can view the workflow [here](https://github.com/arsyun/go-filecoin-storage-helper)
- Yes, such as 
    - storage-helper store [path]
        - --status [cid] // check the storage flowwork status
        - --expire [time] // set expire time
        - --price [FIL] // set price
        - ......
    - storage-helper retrieval
        - ......
- We'll support lotus although lotus was not open source when the proposal was submitted

File: rfp-proposals/ars-go-filecoin-storage-helper.md (hunk 1 above)
Comment (waynewyang) [reply rfp-proposals/ars-go-filecoin-storage-helper.md]: @pooja  yes,we'll maintenance work without additional funding.

**Comment by pooja** [comment]
We've finished the wave 1 grant selection process. Can you please email devgrants@filecoin.org so we can share our decision and next steps, if any? In your email, please include a link to your proposal PR so we know who we're talking to!

Thanks,
Filecoin DevGrants team

**Comment by pooja** [comment]
Greetings,

Thank you for your application to the Filecoin DevGrants program. We appreciate your proposal!

You have been selected for a Wave 1 DevGrant, congratulations! 🎉 

We selected grants based on:
- what makes sense to build now based on where the Filecoin node implementations are
- the likelihood of your team being able to work independently of the Filecoin engineering teams working on the core client implementations
- degree of refactoring your proposal may require as the clients evolve
- the integrity of the proposal and near-term value to the Filecoin ecosystem
- your commitment to the ecosystem and desire to maintain your project longer-term

Please reach out to the DevGrants team at devgrants@filecoin.org with a link to this PR, and we will follow up with next steps.

**Pull request closed without merging**